src/views/ClosePurchaseOrderModal.vue

```diff
@@ -0,0 +1,182 @@
+          <ion-badge v-else-if="item.orderItemStatusId === 'ITEM_REJECTED'" color="danger" slot="end">{{ $t("Rejected") }}</ion-badge>
+          <ion-checkbox v-else slot="end" :modelValue="item.isChecked" />
+        </ion-buttons>
+      </ion-item>
+    </ion-list>
+  </ion-content>
+
+  <ion-fab vertical="bottom" horizontal="end" slot="fixed">
+    <ion-fab-button :disabled="!hasPermission(Actions.APP_SHIPMENT_UPDATE) || !isEligibleToClosePOItems()" @click="confirmSave">
+      <ion-icon :icon="saveOutline" />
+    </ion-fab-button>
+  </ion-fab>
+</template>
+
+<script lang="ts">
+import {
+  alertController,
+  IonButton,
+  IonButtons,
+  IonBadge,
+  IonCheckbox,
+  IonContent,
+  IonFab,
+  IonFabButton,
+  IonHeader,
+  IonIcon,
+  IonItem,
+  IonLabel,
+  IonList,
+  IonListHeader,
+  IonTitle,
+  IonToolbar,
+  IonThumbnail,
+  modalController
+} from '@ionic/vue';
+import { Actions, hasPermission } from '@/authorization'
+import { closeOutline, checkmarkCircle, arrowBackOutline, saveOutline } from 'ionicons/icons';
+import { defineComponent } from 'vue';
+import { mapGetters } from 'vuex'
+import { OrderService } from "@/services/OrderService";
+import { productHelpers, showToast } from '@/utils';
+import { ShopifyImg } from '@hotwax/dxp-components';
+import { translate } from '@/i18n'
+import emitter from "@/event-bus"
+import { useRouter } from 'vue-router';
+
+export default defineComponent({
+  name: "ClosePurchaseOrderModal",
+  components: {
+    IonBadge,
+    IonButton,
+    IonButtons,
+    IonCheckbox,
+    IonContent,
+    IonFab,
+    IonFabButton,
+    IonHeader,
+    IonIcon,
+    IonItem,
+    IonLabel,
+    IonList,
+    IonListHeader,
+    IonTitle,
+    IonThumbnail,
+    IonToolbar,
+    ShopifyImg
+  },
+  computed: {
+    ...mapGetters({
+      getProduct: 'product/getProduct',
+      order: 'order/getCurrent',
+      productIdentificationPref: 'user/getProductIdentificationPref'
+    })
+  },
+  props: ['isEligibileForCreatingShipment'],
+  methods: {
+    closeModal() {
+      modalController.dismiss({ dismissed: true });
+    },
+    async confirmSave() {
+      const alert = await alertController.create({
+        header: this.$t('Close purchase order items'),
+        message: this.$t('Are you sure you have received the purchase order for the selected items? Once closed, the shipments for the selected items wont be available for receiving later.', { space: '<br /><br />' }),
+        buttons: [{
+          text: this.$t('Cancel'),
+          role: 'cancel'
+        },
+        {
+          text: this.$t('Proceed'),
+          role: 'proceed',
+          handler: async() => {
+            if(this.isEligibileForCreatingShipment) {
+              emitter.emit('create-shipment')
+            }
+            await this.updatePOItemStatus()
+            modalController.dismiss()
+            this.router.push('/purchase-orders')
```

Comment:
Why are we changing the route from here and also changing the same from the createShipment method?